cdisc_rules_engine/services/datasetxpt_metadata_reader.py

```diff
@@ -18,40 +15,48 @@ class DatasetXPTMetadataReader:
     # TODO. Maybe in future it is worth having multiple constructors
     #  like from_bytes, from_file etc. But now there is no immediate need for that.
     def __init__(self, file_path: str, file_name: str):
-        with open(file_path, "rb") as f:
-            self._file_contents = f.read()
-        self._metadata_container = None
+        file_size = os.path.getsize(file_path)
+        if file_size > psutil.virtual_memory().available * 0.25:
```

Comment:
Shouldn't this line be more tightly coupled with this line:
https://github.com/cdisc-org/cdisc-rules-engine/blob/a9edcfb26d3235fac825919c6df3f79ad2021f85/cdisc_rules_engine/services/data_services/data_service_factory.py#L94

For example, let's say the dataset is considered too large here and we don't do a count. But then, the memory is gc'd and we later decide that the dataset can be read into ram with pandas. Will we end up without a record count?
In the opposite edge case, we compute the size twice.

Reply:
While we won't ever end up without a record count because even though we don't read the full file we estimate the record count: 
https://github.com/cdisc-org/cdisc-rules-engine/blob/a9edcfb26d3235fac825919c6df3f79ad2021f85/cdisc_rules_engine/services/datasetxpt_metadata_reader.py#L59

This may be useful when looking at #690 

But I do think we can improve this by making a utility method for getting the dataset_size_threshold